src/common/common.types.ts

```diff
@@ -77,6 +83,38 @@ interface Data {
   created_at?: Date;
 }
 
+/**
+ * Type for creating data and sending it to the API.
+ */
+type DataCreate = Required<Pick<Data, "variable">> &
+  Partial<
+    Omit<Data, "id" | "device" | "origin" | "location" | "time" | "created_at"> & {
+      /**
+       * Location for the data value.
+       */
+      location: {
+        /**
+         * Latitude for the data.
+         */
+        lat: number;
+        /**
+         * Latitude for the data.
+         */
+        lng: number;
+      };
```

Comment:
Location has a different format when `POST`ing data. This is also used in the `DataEdit` type below.